### ISSUE TITLE
ModelStatus Unknown but successful after second run.

### ISSUE THREAD

**Issue by timellemeet** [ERGO-Code/HiGHS#1528]
Using `highspy` I run a column generation algorithm where I solve the model many times and check if the solution is optimal. Occasionally I get `HighsModelStatus.kUnknown`, I can't find any issues and if I run the model again (without changing it) it solves just fine. I would love a little bit more documentation about this status so I can further debug what is causing it.

Btw loving the solver!

**Comment by jajhall** [comment]
Strange. I know that there are circumstances when this can happen. What are the numbers of primal and dual infeasibilities?

**Comment by timellemeet** [comment]
What do you mean exactly?

**Comment by jajhall** [comment]
Are you getting kUnknown immediately after a call to run(), or after you've modified the problem and found that it's optimal yourself?

**Comment by timellemeet** [comment]
Immediately after. To be precise:

1. run()
2. getObjectiveValue()
3. .getModelStatus() -> <HighsModelStatus.kUnknown: 15>

If I do run() again right after it returns HighsStatus.kOk and when I get the status it is optimal.

It does seem to happen more when I also start adding constraints, not only columns but that's mainly speculation.

**Comment by jajhall** [comment]
So, after that first call to run(), if you call

Info = h.getInfo()

what are the values of

info.num_primal_infeasibility

... Let me finish this on my laptop

**Comment by timellemeet** [comment]
I see

```basis_validity:1
crossover_iteration_count:0
dual_solution_status:1
ipm_iteration_count:0
max_dual_infeasibility:0.022486178773760912
max_integrality_violation:inf
max_primal_infeasibility:0.0
mip_dual_bound:0.0
mip_gap:inf
mip_node_count:-1
num_dual_infeasibilities:1
num_primal_infeasibilities:0
objective_function_value:243263302.553092
primal_solution_status:2
qp_iteration_count:0
simplex_iteration_count:104
sum_dual_infeasibilities:0.022486178838871496
sum_primal_infeasibilities:0.0
valid:True```

![image](https://github.com/ERGO-Code/HiGHS/assets/6830281/a99caf89-e435-4a58-9a14-b7b55891492b)


If I then do another run:

```basis_validity:1
crossover_iteration_count:0
dual_solution_status:2
ipm_iteration_count:0
max_dual_infeasibility:2.2851054382044822e-11
max_integrality_violation:inf
max_primal_infeasibility:0.0
mip_dual_bound:0.0
mip_gap:inf
mip_node_count:-1
num_dual_infeasibilities:0
num_primal_infeasibilities:0
objective_function_value:243263302.55172285
primal_solution_status:2
qp_iteration_count:0
simplex_iteration_count:122
sum_dual_infeasibilities:1.3606893389805919e-10
sum_primal_infeasibilities:0.0
valid:True```

![image](https://github.com/ERGO-Code/HiGHS/assets/6830281/2679bb3f-1eca-4054-958a-0b64e85a4cf1)

**Comment by jajhall** [comment]
Very odd. I'll get back to you later

**Comment by jajhall** [comment]
Could you also set setOptionValue("log_dev_level", 1) before the first call to run(), and send me the output

**Comment by timellemeet** [comment]
Hopefully this is helpful, the run that gives unknown:

![image](https://github.com/ERGO-Code/HiGHS/assets/6830281/e0b1446f-fabe-4cc8-a555-c3b23448a3ce)

And a run right after:
![image](https://github.com/ERGO-Code/HiGHS/assets/6830281/d51d4c7f-ae81-4f99-830d-162b2fc9eb1d)

**Comment by jajhall** [comment]
Yes, this explains everything. The simplex solver has identified cycling, so failed rather than risk an infinite loop. My advice to you is to re-run it you get this model status - which I guess is what you're doing.

Perhaps we could return a more informative model status 🙂

**Comment by timellemeet** [comment]
Clear, indeed the way to fix it is fine but a dedicated model status would maybe make sense. But if a direct retry solves it, does it not make sense to implement that also? 

Thanks for you help btw!

**Comment by timellemeet** [comment]
One more thought, in the use case of column generation, should I actually do a retry? Or should I just continue adding columns as I create a new model anyway?

**Comment by jajhall** [comment]
This is part of the art of column generation. More resolves cost time, but give you duals with respect to newer columns. I'm not sufficiently expert to comment further
